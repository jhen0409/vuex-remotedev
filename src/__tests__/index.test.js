import Vue from 'vue'
import Vuex from 'vuex'
import plugin from '..'

Vue.use(Vuex)

// Mock redux-devtools-extension API
const init = jest.fn()
const send = jest.fn()
const subscribe = jest.fn(() => jest.fn())
window.devToolsExtension = {
  connect: jest.fn(() => ({
    init,
    send,
    subscribe,
  })),
}

test('works without crashing', () => {
  const store = new Vuex.Store({
    state: {
      count: 1,
    },
    mutations: {
      increment(state) {
        // mutate state
        state.count++
      },
    },
    actions: {
      increment(context, payload) {
        context.commit('increment', payload)
      },
    },
  })
  plugin(store)
  expect(store.state).toMatchSnapshot()
  expect(subscribe.mock.calls).toMatchSnapshot()

  store.dispatch('increment', { a: 1 })
  expect(init.mock.calls).toMatchSnapshot()
  expect(send.mock.calls).toMatchSnapshot()
})
