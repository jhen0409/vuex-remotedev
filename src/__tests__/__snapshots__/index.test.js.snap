// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works without crashing 1`] = `
Object {
  "count": 1,
}
`;

exports[`works without crashing 2`] = `
Array [
  Array [
    [Function],
  ],
]
`;

exports[`works without crashing 3`] = `
Array [
  Array [
    Object {
      "count": 1,
    },
  ],
]
`;

exports[`works without crashing 4`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "a": 1,
      },
      "type": "increment",
    },
    Object {
      "count": 2,
    },
  ],
]
`;
